test/known_problems/should_pass/list_nth.erl

```diff
@@ -0,0 +1,32 @@
+-module(list_tail).
+
+-compile([export_all]).
+
+% Pass
+-spec nth1([integer()]) -> integer().
+nth1([]) -> 0;
+nth1(Xs) -> lists:nth(1, Xs).
+
+% Fail but should pass
+-spec nth2([integer()]) -> integer().
+nth2(Xs) ->
+    case length(Xs) > 0 of
+        true -> lists:nth(1, Xs);
+        false -> 0
+    end.
+
+% Fail but should pass
+-spec nth3([integer()]) -> integer().
+nth3(Xs) ->
+    case Xs of
+        [] -> 0;
+        _ -> lists:nth(1, Xs)
+    end.
```

Comment:
Good one! This is refinement of an already bound variable (bound before the pattern matching) using the mismatching patterns preceding this one, i.e. []. We have a known problem module for this already, so I think this test case should be moved to `test/known_problems/should_pass/refine_bound_var_on_mismatch.erl`.